File: src/ArticlesApp.WebAPI/AppFrontend/src/Entities/Article.ts
import { ArticleStatesEnum } from "src/Entities/ArticleStatesEnum";



export class Article {
  constructor(ar: Article) {
    this.id = ar.id;
    this.title = ar.title;
    this.text = ar.text;
    this.submittedAt_DateUtc = ar.submittedAt_DateUtc;
    this.authorId = ar.authorId;
    this.articleStateId = ar.articleStateId;
    this.articleStateId_LastChangedAt_DateUtc = ar.articleStateId_LastChangedAt_DateUtc;
    this.articleStateId_LastChangedBy_ModeratorId = ar.articleStateId_LastChangedBy_ModeratorId;
    this.softDeletedAt_DateUtc = ar.softDeletedAt_DateUtc;
    this.versionId = ar.versionId;
  }

  id: number;

  title: string;
  text: string;

  submittedAt_DateUtc: Date;

  authorId: string;

  articleStateId: ArticleStatesEnum;

  articleStateId_LastChangedAt_DateUtc?: Date;
  articleStateId_LastChangedBy_ModeratorId?: string;

  softDeletedAt_DateUtc?: Date;

  versionId: number;
}



export type ArticleSubmit = Pick<Article, "title" | "text">;

File: src/ArticlesApp.Core/Entities/Article/Article.cs
using ArticlesApp.Core.Entities.SoftDeletion;



namespace ArticlesApp.Core.Entities.Article;



public class Article : IArticle, ISoftDeletable
{
    public Article(
        int id,
        string title,
        string text,
        DateTime submittedAt_DateUtc,
        string authorId,
        ArticleStatesEnum articleStateId,
        short versionId
    )
    {
        Id = id;
        Title = title;
        Text = text;
        SubmittedAt_DateUtc = submittedAt_DateUtc;
        AuthorId = authorId;
        ArticleStateId = articleStateId;
        VersionId = versionId;
    }

    public int Id { get; set; }

    public string Title { get; set; }
    public string Text { get; set; }

    public DateTime SubmittedAt_DateUtc { get; set; }

    public string AuthorId { get; set; }

    public ArticleStatesEnum ArticleStateId { get; set; }

    public DateTime? ArticleStateId_LastChangedAt_DateUtc { get; set; }
    public string? ArticleStateId_LastChangedBy_ModeratorId { get; set; }

    public DateTime? SoftDeletedAt_DateUtc { get; set; }

    public short VersionId { get; set; }
}